docs/adr/0003-End-to-end-tests.md

```diff
@@ -0,0 +1,57 @@
+# 3. End to end tests
+
+Date: 2023-04-18
+
+## Status
+
+Accepted
+
+## Context
+
+End to end (e2e) tests only run tests against exporters `/metrics` endpoint in OpenShift today (using Prow CI). We want to run them with more cases, like against Grafana, making them more reliable.
+
+## Decision
+
+e2e tests will be more user friendly.
+
+Prow CI will run e2e tests that cover more scenarios and are more reliable.
+
+### How
+
+- We will rewrite the e2e tests script in Python.
+    - easier to write and read it
+    - they will run against Pelorus operator (using test image) and not helm
```

Comment:
They will run against any Pelorus deployment. The deployment method is not part of this ADR. To make it nice and robust we should not combine Pelorus deployment method with e2e tests.

Deployment may be part of the e2e test scripts, but a separate from executing e2e tests itself. The e2e test cases should be smart enough to enable/disable test cases based on the current Pelorus deployment.

Let's say we want to run tests against webhook, deploytime and committime, where committime is has two backends: github, bitbucket.

1. Deploy Pelorus (we don't care if it's Helm or Operator):
    Helm uses: `values.yaml`
    Operator uses: `charts.pelorus.dora-metrics.io/v1alpha1`

2. Verify if the deployment was success.
3. Gather information about deployment (e.g. we have one deploytime, 2 committime with github and bitbucket).
4. Run pre-condition workload based on the data from point 3., such as sending data to the Pelorus via webhook, creating bugs if necessary, deploying application to be monitored by deploytime and so on.
5. Run tests against workload based on the data from point 3. The test cases available may be much higher, but we could dynamically decide if the test is suitable for the particular deployment. 
6. Gather logs from the deployment and test results.

The exit() should happen only after point where the logs are gathered.

Reply:
agreed

changing here to

- it will run against any Pelorus deployment (like Helm, Operator SDK, etc)

is ok for you? or do you want more info?

Reply:
yes